src/core/providers/strategies/helpers/getModalHandlers.ts

```diff
@@ -0,0 +1,29 @@
+import { PendingTransactionsStateManager } from 'core/managers/internal/PendingTransactionsStateManager/PendingTransactionsStateManager';
+import { PendingTransactionsModal } from 'lib/sdkDappCoreUi';
+import { ProviderErrorsEnum } from 'types/provider.types';
+import { createUIElement } from 'utils/createUIElement';
+
+export async function getModalHandlers<T>(props?: {
+  cancelAction?: () => Promise<T> | undefined;
+}) {
+  const modalElement = await createUIElement<PendingTransactionsModal>({
+    name: 'pending-transactions-modal'
+  });
+
+  const eventBus = await modalElement.getEventBus();
+
+  if (!eventBus) {
+    throw new Error(ProviderErrorsEnum.eventBusError);
+  }
+
+  const manager = new PendingTransactionsStateManager(eventBus);
+
+  const onClose = async (shouldCancelAction = true) => {
```

Comment:
Maybe rename this to `handleClose`? We've discussed before that the `on` prefix is usually assigned to when the callback comes as a component property.